### DISCUSSION TITLE
Multivariate processing and Any-variate attention

### DISCUSSION THREAD

**Discussion by icisneros** [SalesforceAIResearch/uni2ts#61] [Q&A]
Hello, and thank you for releasing a very cool codebase!

I am new to time series forecasting, so please forgive my ignorance. I am trying to confirm my understanding of the setup that you have so that I may adapt your model to my use case. In particular I am looking at the implementations in [example/moirai_forecast_pandas.ipynb](https://github.com/SalesforceAIResearch/uni2ts/blob/main/example/moirai_forecast_pandas.ipynb).

Essentially, I would like to feed as input a multivariate time series and ensure that the resulting forecasts are considering all of the variates when forecasting any one of the variates, which I believe you call "Any-variate attention" in the paper.  For example, let's assume I have a 'wide' pandas DataFrame with 100 timestamps and 10 variates (A, B, C, ... J), which are all correlated in some subtle way.  When forecasting time series A, the model should consider the input data from variates B through J.  When forecasting time series B, the model should consider the data from A, C, ... J. Is this being demonstrated in either `2.3 Multivariate Forecasting` or `2.4 Include Dynamic Features`  in the example notebook ?

When probing the models' TransformerEncoder `forward()` method, the example in 2.3 seems to feed in the input as a flattened 1d sequence, and runs a single `forward` call.  
When running the example in 2.4, the input is also a flattened 1d sequence, but there are several sequential `forward` calls, with progressively increasing patch sizes, but I do not understand why.

**Comment by gorold** [comment]
You could just use 2.3 Multivariate Forecast. 

The progressively increasing patch sizes is due to patch_size being set to "auto", and it's doing automatic selection of the patch size hparam.
